spec/abci++/abci++_properties_001_draft.md

```diff
@@ -0,0 +1,186 @@
+---
+order: 1
+title: New Methods
+---
+
+# New Methods
+
+## Description
+
+### PrepareProposal
+
+#### Parameters and Return values
+
+TODO
+
+#### When does Tendermint call it?
+
+When a validator _p_ enters Tendermint consensus round _r_, height _h_, in which _p_ is the proposer:
+
+1. _p_'s Tendermint collects the outstanding transactions from the mempool (TODO: should we limit size & gas?).
+2. _p_'s Tendermint creates a block header.
+3. _p_'s Tendermint calls `PrepareProposal` with the newly created block. The call is synchronous (i.e., Tendermint's execution will block until the Application returns).
+4. The Application checks the block (the header and transactions). It can also:
+   * add/remove transactions
+   * modify the header hashes
+   * if the block is modified, the Application sets it in the return parameters
+5. The Application signals _Accept_ of _Reject_ in `PrepareProposal`'s return values
```

Comment:
```suggestion
5. The Application signals _Accept_ or _Reject_ in `PrepareProposal`'s return values
```

Reply:
Does this mean that a bug in the application ("always reject", or "reject for three weeks") may hold the chain?